pkg/plugins/codegen/jenny_flatten.go

```diff
@@ -0,0 +1,34 @@
+package codegen
+
+import (
+	"github.com/grafana/codejen"
+	corecodegen "github.com/grafana/grafana/pkg/codegen"
+)
+
+func FlattenJenny(gen codejen.OneToOne[*corecodegen.DeclForGen]) codejen.OneToMany[[]*corecodegen.DeclForGen] {
+	return &flattenJenny{
+		inner: gen,
+	}
+}
+
+type flattenJenny struct {
+	inner codejen.OneToOne[*corecodegen.DeclForGen]
+}
+
+func (gen *flattenJenny) JennyName() string {
+	return "FlattenJenny"
+}
+
+func (gen *flattenJenny) Generate(decls []*corecodegen.DeclForGen) (codejen.Files, error) {
+	files := make(codejen.Files, len(decls))
+
+	for _, decl := range decls {
+		file, err := gen.inner.Generate(decl)
+		if err != nil {
+			continue
+		}
+		files = append(files, *file)
```

Comment:
Some warts in codejen system right now - it's OK for a `*ToOne` jenny to return a nil pointer to indicate "nothing to do for this input". This will panic in that case. Also, need to pass errors back up, or else they're swallowed. I'd suggest this: 

```suggestion
		if err != nil {
			return nil, err
		}
		if file != nil && file.Exists() {
			files = append(files, *file)
		}
```